baseline/baseline.py

```diff
@@ -82,79 +105,169 @@ def load_model(model_file_path: str, device: str = "cpu") -> Any:
     return RandomProjectionMelEmbedding().to(device)
 
 
+def frame_audio(audio: Tensor, frame_size: int, hop_size: int, is_centered: bool):
+    """
+    Slice audio into equal length frames. Each adjacent frame is a hop_size number of
+    samples apart.
+
+    Args:
+        audio: input audio, expects a 2d Tensor of shape: (batch_size, num)samples)
+        frame_size: the length each resulting frame should be
+        hop_size: number of samples between frames.
+        is_centered: Pad audio by frame_size // 2 to center audio in frame
+
+    Returns:
+        A Tensor of shape (batch_size, num_frames, frame_size)
+    """
+    batch_size, num_samples = audio.shape
+
+    # Adjust the number of samples if centering to allow for a half
+    # frame size number of padded samples padded at the start.
+    start_pad = 0
+    if is_centered:
+        half_frame_size = int(frame_size // 2)
+        num_samples += half_frame_size
+        start_pad = half_frame_size
```

Comment:
@maxsolomonhenry -- modified this a little bit, we were overpadding non-centered signals so I tried this approach of adjusting the num_samples by half_frame_size if centering prior to calculating the number of samples to append to the end.

Our implementation right now isn't quite in alignment with how librosa handles framing. Since we are padding the end to make sure that the entire frame from the last hop is entirely contained, we are ending up padding more at the end then I think we need to. When looking at the corresponding timestamps we end up with centers of frames that extend beyond the duration of the input. 

The way that librosa handles this is to not zero pad at all when not centering. In this case the input can be truncated a bit. When centering they pad both the start and end by half a frame and then applying framing (again no zero padding occurs during framing).

What we have here works fine, but it is a bit confusing to get a list of timestamps back that extends past the duration of the input.

Reply:
@jorshi Can you we use torchaudio to do this for us?

Reply:
@turian doesn't look like there are any utils for framing in torchaudio

Reply:
@jorshi If we want a centered frame at t = 0 we need to pad up front. As for the end, if you don't want time stamps that extend beyond the signal we'll just have to truncate the output by one frame.

```
if end_pad > half_frame_size:
    # Truncate `audio` instead of padding it.